app/Http/Controllers/DepartmentController.php

```diff
@@ -51,28 +96,86 @@ public function show(string $id)
     /**
      * Show the form for editing the specified resource.
      */
-    public function edit(string $id)
+    public function edit(Request $request, string $id)
     {
-        $department = Department::findOrFail($id);
-        return view('departments.edit', [
-            'department' => $department
-        ]);
+        if($request->user()->isAdmin())
```

Comment:
You shouldn't need to do request for the user, remember you should be using `Auth::`. A request can be manipulated to fool the controller into thinking its an admin, whereas the Auth facade is handled on the backend for us with session management.

Refactor this to use `Auth::user`.

Reply:
resolved in 59ff09f7743b739de37cce6a45b1e3a7053d17fb